### ISSUE TITLE
SYSTEM_THREAD_EXCEPTION_NOT_HANDLED (7e) bugcheck on Win10 1809

### ISSUE THREAD

**Issue by MagicAndre1981** [winsiderss/systeminformer#1523]
### Brief description of your issue

When I download the nightly, expand the zip and run the si exe I get a SYSTEM_THREAD_EXCEPTION_NOT_HANDLED (7e) bugcheck under Windows 10 1809 LTSC

### Steps to reproduce (optional)

- download [nightly](https://system-informer.com/nightly.php)
- run it under Windows 10 1809

### Actual behavior (optional)

SYSTEM_THREAD_EXCEPTION_NOT_HANDLED (7e) bugcheck on Win10 1809, so I compiled my own version and opened the dump, forced the load of my mismatched PDB to get a somehow better callstack:

```
*******************************************************************************
*                                                                             *
*                        Bugcheck Analysis                                    *
*                                                                             *
*******************************************************************************

SYSTEM_THREAD_EXCEPTION_NOT_HANDLED (7e)
This is a very common BugCheck.  Usually the exception address pinpoints
the driver/function that caused the problem.  Always note this address
as well as the link date of the driver/image that contains this address.
Arguments:
Arg1: ffffffffc0000005, The exception code that was not handled
Arg2: fffff8034f47feb0, The address that the exception occurred at
Arg3: ffffd48738eca078, Exception Record Address
Arg4: ffffd48738ec98c0, Context Record Address

EXCEPTION_RECORD:  ffffd48738eca078 -- (.exr 0xffffd48738eca078)
ExceptionAddress: fffff8034f47feb0 (nt!ExpLookupHandleTableEntry)
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000000
NumberParameters: 2
   Parameter[0]: 0000000000000000
   Parameter[1]: 0000000000000032
Attempt to read from address 0000000000000032



PROCESS_NAME:  System

READ_ADDRESS:  0000000000000032 

ERROR_CODE: (NTSTATUS) 0xc0000005 - Die Anweisung in 0x%p verwies auf Arbeitsspeicher bei 0x%p. Der Vorgang %s konnte im Arbeitsspeicher nicht durchgef hrt werden.

EXCEPTION_CODE_STR:  c0000005

EXCEPTION_PARAMETER1:  0000000000000000

EXCEPTION_PARAMETER2:  0000000000000032

EXCEPTION_STR:  0xc0000005
```

### Environment (optional)

- System Informer Rev. 5578
- Windows 10 1809 LTSC 17763.3650 x64 German inside a VMware VM
```

**Comment by jxy-s** [comment]
@MagicAndre1981 that call stack is highly illogical to me. If possible, using the nightly build with the release driver, please reproduce the crash and email me the dump file.

**Comment by jxy-s** [comment]
@MagicAndre1981 that call stack is highly illogical to me. If possible, using the nightly build with the release driver, please reproduce the crash and email me the dump file.

**Comment by MagicAndre1981** [comment]
the crash happens with the release driver, as said, I just downloaded the zip, expanded the zip, tried to run SI and got the BSOD.

**Comment by jxy-s** [comment]
> so I compiled my own version and opened the dump, forced the load of my mismatched PDB to get a somehow better callstack

Sorry, I seem to have misunderstood your message.

Regardless, it looks like a null pointer dereference. Potentially due to bad offsets in dynamic data. Please email me the dump file so I can investigate. Or, using the dump file, try to dump the information on the currently loaded dynamic data:

```cpp
static KPH_ACTIVE_DYNDATA KphpActiveDynData = { 0 };
```

It's hard to tell for sure, given the mismatched symbols, but this may be retrieving the wrong pointer.

https://github.com/winsiderss/systeminformer/blob/master/KSystemInformer/object.c#L78

Dumping the EPROCESS object and KphpActiveDynData will help, if possible, given the bad symbols. Unsure why you can't load them. They're here:

https://github.com/winsiderss/systeminformer/tree/master/KSystemInformer%2Fbin-signed%2Famd64

**Comment by MagicAndre1981** [comment]
I wrote an mail to your live mail account from your profile

**Comment by jxy-s** [comment]
Thanks, I'm away from a computer today. I'll look when I'm back.

**Comment by dmex** [comment]
> Unsure why you can't load them. They're here:

I've had similar issues in the past related to PDBALTPATH stripping the PDB filename from the debug header:

![image](https://user-images.githubusercontent.com/1306177/205759073-5ceb2fbe-6c52-40e6-8532-79347001803b.png)

The other issue is how dbghelp doesn't check the directory where the image is located:

![image](https://user-images.githubusercontent.com/1306177/205758701-bf5cdc97-169b-4435-8165-caebb5a61a4c.png)

PDBALTPATH strips the path and it becomes relative. Sometime later dbghelp tries using the filename but since it's relative ends up in the current directory (the same location as the executable and not allowed) then won't check elsewhere and just doesn't load the PDB 🙄

> forced the load of my mismatched PDB

@MagicAndre1981 

1) Copy the pdbs from \bin-signed\amd64\ into `somefolder` 
2) Set the symbol search path for the debugger using this exact format:
`D:\somefolder;SRV*C:\Symbols*msdl.microsoft.com`

This way you can give proper stack traces that make sense unlike the homebrew mismatched PDB you've been using 😉

**Comment by MagicAndre1981** [comment]
> @MagicAndre1981 I just got back to check my email and I still haven't received an email from you.

confusing. I wrote the mail yesterday. I sent a new one, also check junk folder

**Comment by MagicAndre1981** [comment]
>. Set the symbol search path for the debugger using this exact format:
>    `D:\somefolder;SRV*C:\Symbols*msdl.microsoft.com`
> 
> This way you can give proper stack traces that make sense unlike the homebrew mismatched PDB you've been using wink

here is the output:

```
STACK_TEXT:  
ffffc08d`e4b7d2b8 fffff804`4ff458d2     : 00000000`00000000 00000000`00000000 9a70a28a`86d416bf b55f17de`53678e9d : nt!ExpLookupHandleTableEntry
ffffc08d`e4b7d2c0 fffff804`4ff457e4     : fffff804`55204674 fffff804`4ff57460 ffff9109`0cfbf300 fffff804`5520c4bf : nt!ExpGetNextHandleTableEntry+0x22
ffffc08d`e4b7d2f0 fffff804`55204782     : ffffc08d`e4b7d528 ffffc08d`e4b7d528 fffff804`55207f40 fffff804`551fc0a0 : nt!ExEnumHandleTable+0x44
ffffc08d`e4b7d360 fffff804`55207ed6     : 00000000`00000010 00000000`00010213 ffffc08d`e4b7d528 ffffc08d`e4b7d528 : SystemInformer!KphEnumerateProcessHandlesEx+0x10e [C:\Users\aione\source\repos\systeminformer\KSystemInformer\object.c @ 273] 
ffffc08d`e4b7d3c0 fffff804`551fc0f3     : ffffc08d`e4b7d4c0 00000000`3363704b ffffc08d`e4b7d499 fffff804`4f8e281b : SystemInformer!KphEnumProcessContextsForProtection+0x36 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\protection.c @ 391] 
ffffc08d`e4b7d3f0 fffff804`551faeeb     : ffffc70b`17135098 00000000`00000000 ffffc08d`00000000 fffff804`4fefd07b : SystemInformer!KphpEnumerateContexts+0x53 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_tracking.c @ 1280] 
ffffc08d`e4b7d420 fffff804`551fae14     : ffffc70b`19549000 00000000`00000000 00000000`00000013 ffffc70b`1717fd50 : SystemInformer!CidEnumerateInvokeCallback+0x6b [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_table.c @ 617] 
ffffc08d`e4b7d450 fffff804`551fbd03     : ffffc70b`1717fd01 fffff804`551fc0a0 00000000`00000000 00000000`0012184b : SystemInformer!CidEnumerateInternal+0xe8 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_table.c @ 714] 
ffffc08d`e4b7d4a0 fffff804`5520826f     : ffffc70b`1717fd01 00000000`0012184b 00000000`00000000 fffff804`4f9ce8c3 : SystemInformer!KphEnumerateProcessContexts+0x3f [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_tracking.c @ 1305] 
ffffc08d`e4b7d500 fffff804`551fbf2b     : ffffc70b`1717fd50 ffffc70b`1717fd50 fffff804`551fc0a0 e8c99d43`00000000 : SystemInformer!KphStartProtectingProcess+0x12f [C:\Users\aione\source\repos\systeminformer\KSystemInformer\protection.c @ 518] 
ffffc08d`e4b7d590 fffff804`551fc0f3     : ffffc70b`1717fd50 00000000`0000001e 00000000`00000001 fffff804`4fe0e100 : SystemInformer!KphpCidEnumPostPopulate+0x3b [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_tracking.c @ 771] 
ffffc08d`e4b7d5c0 fffff804`551faeeb     : ffffc70b`166f7330 00000000`00000000 01000000`00100000 fffff804`4f8ea8e7 : SystemInformer!KphpEnumerateContexts+0x53 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_tracking.c @ 1280] 
ffffc08d`e4b7d5f0 fffff804`551fae14     : ffffc70b`19549010 00000000`00000002 00000000`00000066 00000000`00000040 : SystemInformer!CidEnumerateInvokeCallback+0x6b [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_table.c @ 617] 
ffffc08d`e4b7d620 fffff804`551fbcaf     : 00000000`00000103 fffff804`551fc0a0 00000000`00000000 ffffc08d`e4b7d730 : SystemInformer!CidEnumerateInternal+0xe8 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_table.c @ 714] 
ffffc08d`e4b7d670 fffff804`551fbc41     : 00000000`00000103 fffff804`551f7000 00000000`00000b58 fffff804`3163704b : SystemInformer!KphEnumerateCidContexts+0x3f [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_tracking.c @ 1355] 
ffffc08d`e4b7d6d0 fffff804`55203a0d     : 00000000`00000000 ffff9109`0ad23000 00000000`00000000 00000000`00000000 : SystemInformer!KphCidPopulate+0x66d [C:\Users\aione\source\repos\systeminformer\KSystemInformer\cid_tracking.c @ 1080] 
ffffc08d`e4b7d760 fffff804`55210020     : 00000000`00000000 ffff9109`0a64ae10 ffff9109`0ad23000 ffff30c5`507a890a : SystemInformer!DriverEntry+0x555 [C:\Users\aione\source\repos\systeminformer\KSystemInformer\main.c @ 300] 
ffffc08d`e4b7d920 fffff804`4ff82235     : 00000000`00000000 00000000`00000000 ffff9109`0a64ae10 00000000`00001000 : SystemInformer!GsDriverEntry+0x20 [minkernel\tools\gs_support\kmodefastfail\gs_driverentry.c @ 47] 
ffffc08d`e4b7d950 fffff804`4ff818fa     : ffff9109`048f4180 00000000`00000000 00000000`00000000 ffff9109`03b24050 : nt!IopLoadDriver+0x4bd
ffffc08d`e4b7db30 fffff804`4f96567a     : ffff9109`00000000 ffffffff`800029b8 ffff9109`0a58a040 ffff9109`00000000 : nt!IopLoadUnloadDriver+0x4a
ffffc08d`e4b7db70 fffff804`4f9e7375     : ffff9109`0a58a040 ffff9109`03a7a080 ffff9109`0a58a040 00000065`b8bbbdff : nt!ExpWorkerThread+0x16a
ffffc08d`e4b7dc10 fffff804`4fa759ec     : ffffa600`7bd40180 ffff9109`0a58a040 fffff804`4f9e7320 00000000`00000000 : nt!PspSystemThreadStartup+0x55
ffffc08d`e4b7dc60 00000000`00000000     : ffffc08d`e4b7e000 ffffc08d`e4b78000 00000000`00000000 00000000`00000000 : nt!KiStartSystemThread+0x1c

FAULTING_SOURCE_LINE:  C:\Users\aione\source\repos\systeminformer\KSystemInformer\object.c

FAULTING_SOURCE_FILE:  C:\Users\aione\source\repos\systeminformer\KSystemInformer\object.c

FAULTING_SOURCE_LINE_NUMBER:  273

FAULTING_SOURCE_CODE:  
No source found for 'C:\Users\aione\source\repos\systeminformer\KSystemInformer\object.c'


SYMBOL_NAME:  SystemInformer!KphEnumerateProcessHandlesEx+10e

MODULE_NAME: SystemInformer

IMAGE_NAME:  SystemInformer.sys

IMAGE_VERSION:  3.2.0.0

STACK_COMMAND:  .cxr 0xffffc08de4b7c8c0 ; kb

BUCKET_ID_FUNC_OFFSET:  10e

FAILURE_BUCKET_ID:  AV_SystemInformer!KphEnumerateProcessHandlesEx

OS_VERSION:  10.0.17763.1

BUILDLAB_STR:  rs5_release

OSPLATFORM_TYPE:  x64

OSNAME:  Windows 10
```

**Comment by jxy-s** [comment]
Thanks for emailing me the dump file. I've identified why dynamic data is not being loaded correctly. We have a plan to address this.

**Comment by gettysburg** [comment]
Hello,

I encounter the same bugcheck as soon as I check the "Enable kernel-mode driver" checkbox under Settings.

OS: Windows Server 2019 LTSC - Version 1809 - 17763.3650
SystemInformer Build: Nightly 3.0.5578

A solution would be much appreciated.

**Comment by BK8000L** [comment]
https://disk.yandex.com/d/I932kx8-bmhXUA
this from my downloads folder version works

**Comment by MagicAndre1981** [comment]
@dmex does "third party" label mean this is a Windows bug in 1809/Server 2019?

**Comment by jxy-s** [comment]
> @dmex does "third party" label mean this is a Windows bug in 1809/Server 2019? 

No, we're waiting for Microsoft. The holidays are slowing down response times in our communication with them.

**Comment by Anubioz** [comment]
Can confirm this thing happening on Windows 2019 Server too. SystemInformers kernel-mode driver caused BSOD for me since it was released as an update to ProcessHacker (which in turn worked like a charm with kernel driver enabled). Gonna try to build it myself to get more info

**Comment by Anubioz** [comment]
So here is my dump:

```!analyze -v
*******************************************************************************
*                                                                             *
*                        Bugcheck Analysis                                    *
*                                                                             *
*******************************************************************************

SYSTEM_THREAD_EXCEPTION_NOT_HANDLED_M (1000007e)
This is a very common bugcheck.  Usually the exception address pinpoints
the driver/function that caused the problem.  Always note this address
as well as the link date of the driver/image that contains this address.
Some common problems are exception code 0x80000003.  This means a hard
coded breakpoint or assertion was hit, but this system was booted
/NODEBUG.  This is not supposed to happen as developers should never have
hardcoded breakpoints in retail code, but ...
If this happens, make sure a debugger gets connected, and the
system is booted /DEBUG.  This will let us see why this breakpoint is
happening.
Arguments:
Arg1: ffffffffc0000005, The exception code that was not handled
Arg2: fffff800727ebfd0, The address that the exception occurred at
Arg3: ffffa18870653068, Exception Record Address
Arg4: ffffa188706528b0, Context Record Address

Debugging Details:
------------------


KEY_VALUES_STRING: 1

    Key  : AV.Dereference
    Value: NullClassPtr

    Key  : AV.Fault
    Value: Read

    Key  : Analysis.CPU.mSec
    Value: 5640

    Key  : Analysis.DebugAnalysisManager
    Value: Create

    Key  : Analysis.Elapsed.mSec
    Value: 20770

    Key  : Analysis.Init.CPU.mSec
    Value: 7405

    Key  : Analysis.Init.Elapsed.mSec
    Value: 84135

    Key  : Analysis.Memory.CommitPeak.Mb
    Value: 83

    Key  : Dump.Attributes.InsufficientDumpfileSize
    Value: 1

    Key  : Dump.Attributes.RequiredDumpfileSize
    Value: 0x5d1aa065


DUMP_FILE_ATTRIBUTES: 0xc
  Insufficient Dumpfile Size
  Kernel Generated Triage Dump

BUGCHECK_CODE:  7e

BUGCHECK_P1: ffffffffc0000005

BUGCHECK_P2: fffff800727ebfd0

BUGCHECK_P3: ffffa18870653068

BUGCHECK_P4: ffffa188706528b0

EXCEPTION_RECORD:  ffffa18870653068 -- (.exr 0xffffa18870653068)
ExceptionAddress: fffff800727ebfd0 (nt!ExpLookupHandleTableEntry)
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000000
NumberParameters: 2
   Parameter[0]: 0000000000000000
   Parameter[1]: 0000000000000032
Attempt to read from address 0000000000000032

CONTEXT:  ffffa188706528b0 -- (.cxr 0xffffa188706528b0)
rax=fffff800bc587eb0 rbx=0000000000000000 rcx=0000000000000032
rdx=0000000000000004 rsi=fffff800bc587f00 rdi=ffffa703ef8d3080
rip=fffff800727ebfd0 rsp=ffffa188706532a8 rbp=0000000000000032
 r8=ffffa18870653310  r9=0000000000000004 r10=ffffa18870653310
r11=ffffa188706533a0 r12=ffffa703eb9ab040 r13=0000000000000000
r14=ffffa18870653378 r15=fffff800bc5848e0
iopl=0         nv up ei pl zr na po nc
cs=0010  ss=0018  ds=002b  es=002b  fs=0053  gs=002b             efl=00010246
nt!ExpLookupHandleTableEntry:
fffff800`727ebfd0 8b01            mov     eax,dword ptr [rcx] ds:002b:00000000`00000032=????????
Resetting default scope

BLACKBOXBSD: 1 (!blackboxbsd)


CUSTOMER_CRASH_COUNT:  1

PROCESS_NAME:  System

READ_ADDRESS: GetUlongFromAddress: unable to read from fffff800726d8240
fffff80072753390: Unable to get MiVisibleState
Unable to get NonPagedPoolStart
Unable to get NonPagedPoolEnd
Unable to get PagedPoolStart
Unable to get PagedPoolEnd
 0000000000000032 

ERROR_CODE: (NTSTATUS) 0xc0000005 - The instruction at 0x%p referenced memory at 0x%p. The memory could not be %s.

EXCEPTION_CODE_STR:  c0000005

EXCEPTION_PARAMETER1:  0000000000000000

EXCEPTION_PARAMETER2:  0000000000000032

EXCEPTION_STR:  0xc0000005

STACK_TEXT:  
ffffa188`706532a8 fffff800`728a4952     : 00000000`01000002 fffff800`bc57c010 00000000`00000010 fffff800`bc5869b2 : nt!ExpLookupHandleTableEntry
ffffa188`706532b0 fffff800`728a4864     : ffffa188`706532f8 fffff800`728b64e0 ffffa703`ef8d3080 fffff800`bc58c3a9 : nt!ExpGetNextHandleTableEntry+0x22
ffffa188`706532e0 fffff800`bc5848a3     : ffffa188`70653518 ffffa188`70653518 fffff800`bc587f50 00000000`00000000 : nt!ExEnumHandleTable+0x44
ffffa188`70653350 fffff800`bc587ee6     : 00000000`00000010 00000000`00010246 ffffa188`70653518 ffffa188`70653518 : SystemInformer+0x148a3
ffffa188`706533b0 fffff800`bc57c064     : ffffa188`706534b0 00000000`3363704b ffffa188`70653489 fffff800`7224181b : SystemInformer+0x17ee6
ffffa188`706533e0 fffff800`bc57ae94     : ffffe707`38cfa0b0 00000000`00000000 ffffa188`00000000 fffff800`7285c0fb : SystemInformer+0xc064
ffffa188`70653410 fffff800`bc57adbc     : ffffe707`38cf0000 00000000`00000000 00000000`00000016 ffffe707`4fdab330 : SystemInformer+0xae94
ffffa188`70653440 fffff800`bc57bc7f     : ffffe707`4fdab301 fffff800`bc57c010 00000000`00000000 00000000`0012184b : SystemInformer+0xadbc
ffffa188`70653490 fffff800`bc588289     : ffffe707`4fdab301 00000000`0012184b 00000000`00000000 fffff800`7232d8c3 : SystemInformer+0xbc7f
ffffa188`706534f0 fffff800`bc57beab     : ffffe707`4fdab330 ffffe707`4fdab330 fffff800`bc57c010 fffff800`bc57c010 : SystemInformer+0x18289
ffffa188`70653580 fffff800`bc57c064     : 00000000`00000010 00000000`00000206 ffffa188`706535d8 00000000`00000018 : SystemInformer+0xbeab
ffffa188`706535b0 fffff800`bc57ae94     : ffffe707`38df0ac8 00000000`00000000 01000000`00100000 fffff800`722498e7 : SystemInformer+0xc064
ffffa188`706535e0 fffff800`bc57adbc     : ffffe707`38cf0028 00000000`00000005 00000000`00000159 00000000`00000040 : SystemInformer+0xae94
ffffa188`70653610 fffff800`bc57bc2b     : 00000000`00000103 fffff800`bc57c010 00000000`00000000 ffffa188`70653730 : SystemInformer+0xadbc
ffffa188`70653660 fffff800`bc57bbbe     : 00000000`00000103 fffff800`bc577000 00000000`00000b58 ffffa703`3163704b : SystemInformer+0xbc2b
ffffa188`706536c0 fffff800`bc583b66     : 00000000`00000000 ffffa703`f9e71000 00000000`00000000 00000000`00000000 : SystemInformer+0xbbbe
ffffa188`70653760 fffff800`bc590020     : 00000000`00000000 ffffa703`f77c6e30 ffffa703`f9e71000 ffff50de`39593ee2 : SystemInformer+0x13b66
ffffa188`70653920 fffff800`728e12a5     : 00000000`00000000 00000000`00000000 ffffa703`f77c6e30 00000000`00001000 : SystemInformer+0x20020
ffffa188`70653950 fffff800`728e096a     : ffffa703`ebf59d50 00000000`00000000 00000000`00000000 ffffa703`e5283ce0 : nt!IopLoadDriver+0x4bd
ffffa188`70653b30 fffff800`722c467a     : ffffa703`00000000 ffffffff`80002050 ffffa703`eb9ab040 ffffa703`00000000 : nt!IopLoadUnloadDriver+0x4a
ffffa188`70653b70 fffff800`72346375     : ffffa703`eb9ab040 ffffa703`e5284040 ffffa703`eb9ab040 00000000`00003971 : nt!ExpWorkerThread+0x16a
ffffa188`70653c10 fffff800`723d49ec     : fffff800`700a2180 ffffa703`eb9ab040 fffff800`72346320 00000000`0199e91c : nt!PspSystemThreadStartup+0x55
ffffa188`70653c60 00000000`00000000     : ffffa188`70654000 ffffa188`7064e000 00000000`00000000 00000000`00000000 : nt!KiStartSystemThread+0x1c


SYMBOL_NAME:  SystemInformer+148a3

MODULE_NAME: SystemInformer

IMAGE_NAME:  SystemInformer.sys

IMAGE_VERSION:  3.2.0.0

STACK_COMMAND:  .cxr 0xffffa188706528b0 ; kb

BUCKET_ID_FUNC_OFFSET:  148a3

FAILURE_BUCKET_ID:  AV_SystemInformer!unknown_function

OSPLATFORM_TYPE:  x64

OSNAME:  Windows 10

FAILURE_ID_HASH:  {6f58ee4d-5f32-1513-3343-451d8a32cb68}

Followup:     MachineOwner
---------
```

**Comment by Anubioz** [comment]
So my bet this is somehow related to `_Acquires_lock_(Process)` `_Releases_lock_(Process)`

**Comment by jxy-s** [comment]
Commit to address this is on the dyndata-fix branch here: https://github.com/winsiderss/systeminformer/commit/496a0bdf5ee9641a84cb819162c26a5462266222

If you'd like to help verify the fix, please cherry-pick that, don't forget to build the driver.

Again, we are blocked by Microsoft at the moment. Hoping to get a resolution here mid-january.

**Comment by MagicAndre1981** [comment]
> Hoping to get a resolution here mid-january.

any response from Microsoft? I still get the bugcheck with latest Patch [17763.3887](https://support.microsoft.com/en-us/topic/january-10-2023-kb5022286-os-build-17763-3887-48683103-7b22-4f36-aa98-0049c7a6e579)

**Comment by gettysburg** [comment]
Just my stoned thought, but we have nightly builds right? This would have been an ideal chance to the users and community to contribute.

In general, nightly builds right now are just like beta builds when compared withh other software, the last "nightly" is over from over 90 days ago.

Just make the current nightly the beta branch, and make an actual nightly that is updated as needed / at least every week.

🚬🌿

**Comment by gettysburg** [comment]
@Anubioz 

Thank you for actually taking the time to try it out! 

There are a few statistics in showing the string "no driver" if the driver is not loaded or working, here is one you could check:

Open System Explorer with it's driver enabled and loaded, then go to it's main window, click "System Information", switch from "CPU" to "Memory", in the Paged Pool box, the limit should be showing accurately if the driver works fine on your system.

Here is the same window, but on 1809 where the driver would bug-check the system:

![image](https://user-images.githubusercontent.com/2256003/214115087-99545eb4-83a2-4b08-ad17-f4ca5442f881.PNG)

If the limit shows accurately for you, then maybe it's time to merge the branch into main?

@jxy-s @dmex

**Comment by jxy-s** [comment]
> If the limit shows accurately for you, then maybe it's time to merge the branch into main?

I can't merge the changes until we get resolution from Microsoft. Since doing so will orphan the existing driver/client ABI. We touched base with Microsoft last week, hoping for resolution soon. Apologies for the delay - engaging with Microsoft can just take a while.

**Comment by dmex** [comment]
>  the limit should be showing accurately if the driver works fine on your system.

@gettysburg Those statistics require symbols and are not available unless you've also installed the Windows SDK.

> Open System Explorer with it's driver enabled and loaded

You don't need to open the sysinfo window to check the driver. The main application window title includes the + symbol after the username to show the driver is enabled and loaded. 

> In general, nightly builds right now are just like beta builds when compared withh other software, the last "nightly" is over from over 90 days ago.

The nightly branch is not stable enough for beta. The last build was 30 days ago on the 25th of December and were on hold for the holidays but have resumed.

> on 1809 where the driver would bug-check

Are you able to update to LTSC 2021? This bug only affects legacy versions of Windows 10 while newer versions don't have issues.

**Comment by MagicAndre1981** [comment]
> Are you able to update to LTSC 2021? This bug only affects legacy versions of Windows 10 while newer versions don't have issues.

I run the 1809 on purpose. It is the Windows 10 which Microsoft sets now as minimum supported system for a lot of new things like WinUI/WinAppSDK. And what I normally do is the start ProcessHacker/SysInformer directly at start in case the system gets unresponsive to be able to easily see what is going on (on my dads laptop with 1904x, search had mem leak and causes unresponsive system because it ran out of memory and it took 30m to start TaskMgr) . And this 1809 is the last LTSC with 10 years support, later[LTSC like the [2021 only get 5 years till 2027](https://learn.microsoft.com/en-us/lifecycle/products/windows-10-enterprise-ltsc-2021), so this [1809 version gets support till 2029 (10 years after release)](https://learn.microsoft.com/en-us/lifecycle/products/windows-10-enterprise-ltsc-2019).

So this 1809 testing VM will run a long time.

**Comment by Anubioz** [comment]
> Are you able to update to LTSC 2021? This bug only affects legacy versions of Windows 10 while newer versions don't have issues.

This bug affects current version of Windows Server 2019, which will be supported until Jan 9, 2029.

**Comment by gettysburg** [comment]
@dmex I chose Server 2019 for all our workstations because of it's long update cycle, the official date when support ends is January 9th, 2029, but some MS officials have said it might be extended to 2031 if enough people still use it by 2029.

Also no, no (official) way to upgrade, either by ISO or from Server Manager, so I'm stuck with 1809.

**Comment by gettysburg** [comment]
> @gettysburg Those statistics require symbols and are not available unless you've also installed the Windows SDK.

Weird, I have the Windows SDK set up correctly as well as the global symbol path, and symbols work for me out of the box in IDA 7.7, as well as other tools utilizing `DbgHelp.dll`.

Regarding an update, I checked and it seems you can't upgrade without having to reconfigure the entire system, there is no ISO or option in Server Manager to do an in-place upgrade.

We chose 2019 LTSC because of the long support cycle, lasting up to 2029, and with some luck, that will be extended by 2 years or they will package and deploy ESU's just like they did on Windows 7.

**Comment by gettysburg** [comment]
> @gettysburg Those statistics require symbols and are not available unless you've also installed the Windows SDK.

Weird, I have the Windows SDK set up correctly as well as the global symbol path, and symbols work pretty much out of the box in IDA 7.7, as well as other tools utilizing `DbgHelp.dll`.

Regarding an update, I checked and it seems you can't upgrade without having to reconfigure the entire system, there is no ISO or option in Server Manager to do an in-place upgrade.

We chose 2019 LTSC because of the long support cycle, lasting up to 2029, and with some luck, that will be extended by 2 years or they will package and deploy ESU's just like they did on Windows 7.

P.S: Sorry for the repeat notifications, if there were any.

**Comment by stdedos** [comment]
From https://github.com/winsiderss/systeminformer/discussions/1580:

> The crash only affects people running old versions of Windows 10 from more than 5 years ago (1809 and older).

and

> https://github.com/winsiderss/systeminformer/files/10521684/minidump.zip

Microsoft Windows [Version 10.0.18363.836]

... also, honestly, I'm not planning to exactly update - when UI changes became "mandatory" with the bug/security fixes, I decided that I like how my OS looks like.
And restarting my PC when I like it, not when Microsoft says so 😕

**Comment by dmex** [comment]
We've decided to temporarily change the minimum version to Windows 10 20H1 (10.0.19041) or newer until such time Microsoft resolve the issue. This means the kernel driver and the majority of features will be disabled unless you're running 20H1 or later.

**Comment by MagicAndre1981** [comment]
> This means the kernel driver and the majority of features will be disabled unless you're running 20H1 or later.

which features in detail? Maybe the team can add a Wiki page with some documentation about which features depends on the driver and which features are available on which Windows.

**Comment by gettysburg** [comment]
@DavidXanatos Most kernel structures are already public through the Microsoft symbol server, and if you have an IDA License and connect to Lumina, you get pretty much everything you need from IDA's public server.

But yes, that's likely whats causing the delay.

**Comment by gettysburg** [comment]
Any news?

Mid January was what you guys told us, now it almost 2 months later.

I rely on the kernel driver on some server machines..

**Comment by jxy-s** [comment]
The delays were escalated mid-February and we got things moving again. I received an email yesterday that we should be good to go this week. However, there are still some administrative items to attend to with respect to portal registration. I don't have a timeline for that. Hopeful that it will all get resolved soon.

I appreciate everyone's patience here. The delays have simply been out of our control. Microsoft has been slow, as somewhat expected, unfortunately. But, they've been understanding and have been working with us to resolve the issues. Kudos to the folks over there for their time.

**Comment by MagicAndre1981** [comment]
ok, great to hear that there is some progress to resolve the issue

**Comment by gettysburg** [comment]
@stdedos I agree with you, they tend to not disclose some stuff, I too have tried in the past, but a resolution this or next week would be wonderful.

**Comment by gettysburg** [comment]
@stdedos I agree with you, they tend to not disclose some stuff, I too have tried in the past - however, a resolution this or next week sounds wonderful.

Good luck to everyone!

**Comment by gettysburg** [comment]
Is this fixed yet?

I am running 3.0.6397 (69f839c) but am hesitant to try it, last time I BSOD'd my system I was left with severe file corruption and had to re-install the OS.

**Comment by gettysburg** [comment]
Is this fixed yet?

I am running the latest version, 3.0.6397 (69f839c), but am hesitant to try loading the driver, last time I did and BSOD'd my system I was left with severe file corruption and had to re-install the OS - good thing I encrypt and sync everything important using a cloud service, but still..

An official statement would be nice, it has been 3 weeks yet again.. I know this is likely Microsoft's fault, but still.

**Comment by gettysburg** [comment]
Is this fixed yet?

Currently running the latest version, 3.0.6397 (69f839c), but I am hesitant to try loading the driver, last time I did and BSOD'd my system I was left with file corruption and had to re-install the OS - good thing I encrypt and sync everything important using a cloud service.

An official statement would be nice, it has been 3 weeks again.. I know this is likely Microsoft's fault, but I am still hoping for good news.

**Comment by dmex** [comment]
> this is likely Microsoft's fault

**Comment by jxy-s** [comment]
Still waiting - see this statement: https://github.com/winsiderss/systeminformer/issues/1523#issuecomment-1458528637

This logic will notify if you're running on an OS that would encounter the bug: https://github.com/winsiderss/systeminformer/blob/master/SystemInformer/ksisup.c#L554

**Comment by dmex** [comment]
> likely Microsoft's fault

That's an understatement.

> Especially helping with System Informer, and internal Kernel stuff, feels very "unexpected"

Well... Zero kelvin is `−273.15°C` and defined as absolute zero and Microsoft are `-273.14°C` which is a slight improvement but still frozen or frosty and nothings changed 😄 

> BSOD'd my system I was left with file corruption and had to re-install the OS

Did you actually have file corruption or did you just see the automatic repair prompt?
https://user-images.githubusercontent.com/1306177/228565683-1bae541f-aab6-4437-9f6d-ba3495eb90fa.png

>  I encrypt and sync everything important using a cloud service.

Is this a virtual drive or physical (IDE/SATA/NVME)?

> I am hesitant to try loading the driver

These crashes only affect legacy versions of Windows 10 prior 2018 and we made changes to prevent the driver running on legacy version of Windows until the issue is resolved 👍

**Comment by DavidXanatos** [comment]
I also notices the driver bricking a freshly installed windows 10 LTSC 2019 and yea the file system was garbage after wards, was less work to reinstall it than to try to find out which file goes where...

**Comment by gettysburg** [comment]
@dmex 

Actual file corruption as confirmed by `chkdsk`, it tried it's best to do an offline repair but the Workstation would behave oddly even after a run of

`sfc /scannow`

and

```
DISM /Online /Cleanup-Image /CheckHealth
DISM /Online /Cleanup-Image /ScanHealth
DISM /Online /Cleanup-Image /RestoreHealth
```

with no results, so I just reinstalled, syncing the data back only takes around 10 minutes or so, the most annoying part was that I lost my custom group policy too, which turned unneeded garbage such as Windows Defender and Firewall off, and much more, should backup and sync that as well.

> Is this a virtual drive or physical (IDE/SATA/NVME)?

Actual physical SSD connected by SATA to my main-boards disk controller.

An idea to make file corruption less likely, perhaps, would be to open device manager, and completely turn off the write-caching for all physical drives when BSOD's or random power outages are likely, but I don't know if this just disables the software / RAM buffers or actually talks to the disk controller using the right IOCTL and disables the hardware write-buffer most drives have nowadays.

I should probably reverse engineer that.. would be interesting to know.

**Comment by gettysburg** [comment]
Any updates? 

It's been another month and I am starting to develop a hatred for Microsoft, even considered switching back to ArchLinux on all my servers and then use QEMU/KVM along with Windows Server 2019 / 2022 Core if Windows is really needed for specific services.

**Comment by jxy-s** [comment]
We're waiting on a permanent fix from Microsoft. They unblocked us and we could sign existing submissions but new ones are getting denied.

**Comment by gettysburg** [comment]
What a joke.

**Comment by jxy-s** [comment]
Staging branch was merged. New driver release soon™️. I greatly appreciate everyone's patience.

**Comment by stdedos** [comment]
Apologies for my pedantic-ness:

1. What was the issue? (e.g. accessing Process handles)
2. What is the solution? (e.g. accessing the API in a Microsoft-sanctioned way)
3. Does the version-limiting apply still?
   1. If yes, does it totally and fully disable the kernel driver unless `v < 20H1`?

**Comment by MagicAndre1981** [comment]
tried to compile it on my own, first `stacktrace` was not found, so add `C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include` to additional includes, but now I get error `STL1001: Unexpected compiler version, expected MSVC 19.34 or newer.` but I use 17.4 LTSC

**Comment by jxy-s** [comment]
> Apologies for my pedantic-ness:
> 
> 1. What was the issue? (e.g. accessing Process handles)
> 2. What is the solution? (e.g. accessing the API in a Microsoft-sanctioned way)

That's a separate issue, but the fix is included. It's described here https://github.com/winsiderss/systeminformer/issues/1525#issuecomment-1345621261

> 3. Does the version-limiting apply still?
>    1. If yes, does it totally and fully disable the kernel driver unless `v < 20H1`?

No, all Win10 x64/ARM64 are supported.

**Comment by MagicAndre1981** [comment]
compile also fails in VisualStudio 2022 Preview 17.6.0-pre.6.0+33626.354, here with this error message:

`error C2220: the following warning is treated as an error [C:\Github\systeminformer\kphlib\kphlib_um.vcxproj]C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.36.32530\include\xlocale(1167,1): warning C5252: Multiple different types resulted in the same XFG type-hash D275361C54538B70; the PDB will only record information for one of them [C:\Github\systeminformer\kphlib\kphlib_um.vcxproj]`

I'll wait for the next official nightly.

**Comment by MagicAndre1981** [comment]
compiling works for debug on the 17.6 Preview6 and looks like it runs now on 1809:

![image](https://user-images.githubusercontent.com/8274816/237001364-4dadfca1-29dd-40ee-9f4a-8c6ab67dd57c.png)

**Comment by stdedos** [comment]
> > 3. Does the version-limiting apply still?
> >    
> >    1. If yes, does it totally and fully disable the kernel driver unless `v < 20H1`?
> 
> No, all Win10 x64/ARM64 are supported.

I'm on the updater-latest version (cd8e90be2858d37dd3667828833f0d80a5b99c4d. 3.0.6550), and I am receiving
![image](https://github.com/winsiderss/systeminformer/assets/133706/030e4b4e-7b17-4b39-8f34-d51452d08784)

Is something else wrong, or am I missing something?

**Comment by dmex** [comment]
@stdedos 

3.0.6550 doesn't include the changes? There hasn't been a nightly build released since 3.0.6550 as a precaution while we investigate #1668

**Comment by MagicAndre1981** [comment]
Version 3.0.6772 (a7cce79) runs fine on 1809, I see the driver loaded, the [++ in Titlebar to indicate full driver usage](https://github.com/winsiderss/systeminformer/blob/a7cce791b342c74168c52abbcd8b597c310c9479/SystemInformer/mainwnd.c#L297)

![image](https://github.com/winsiderss/systeminformer/assets/8274816/8617aad4-28a0-4b2d-adab-57ab5ef4f4c9)

So the issue can be finally closed 🥳🙌😍
